sh/tests/grades_test.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
```

Comment:
The test is not checking for good behavior when more or less arguments than expected are passed.

Also in the current state there is a lot of code duplication and the tests are a bit hard to read.
It would be nice to generalize `challenge` function in order to accept any input and make the `diff` without having to fork for specific case scenarios.

In case you feel it could be useful I've refactored it to have an idea on what the shape could be:
```sh
#!/usr/bin/env bash

IFS='
'
script_dirS=$(cd -P "$(dirname "$BASH_SOURCE")" &>/dev/null && pwd)

challenge() {
    args=${@:1:$#-1}
    input=${@: -1}

    submitted=$(
        bash -c ""$script_dirS"/student/grades.sh $args" <<EOF
$input
EOF
    )
    expected=$(
        bash -c ""$script_dirS"/solutions/grades.sh $args" <<EOF
$input
EOF
    )
    diff <(echo "$submitted") <(echo "$expected")
}

challenge 3 "Bob
90
Alice
150
Eve
55
"

challenge 3 "Bob
90
Alice
100
Eve
55
"

challenge ""

challenge 1 121 21 3 "Alice
1
"
```

Of course you will still need to add the other tests I didn't include into the example.

Reply:
done